Tasks/VsTestPlatformToolInstallerV1/nugetfeedinstaller.ts

```diff
@@ -103,9 +126,52 @@ export class NugetFeedInstaller {
         }
 
         vstestPlatformInstalledLocation = await new NugetDownloadHelper(this.consolidatedCiData)
-            .attemptPackageDownload(packageSource, testPlatformVersion);
+            .attemptPackageDownload(packageSource, testPlatformVersion, nugetConfigFilePath);
 
         // Set the vstest platform tool location for the vstest task to consume
         helpers.setVsTestToolLocation(vstestPlatformInstalledLocation);
     }
+
+    // Utility function that writes the feed url along with username and password if provided into the specified nuget config file
+    private prepareNugetConfigFile(packageSource: string, configFilePath: string, username: string, password: string) : string {
```

Comment:
is this function exists in common lib? if not why aren't we adding there (feel this is common for every nuget end user)

Reply:
nuget task uses a slightly different mechanism. Plus taking a dependency on their code adds too much of a size to ours. wanted to avoid hence wrote my own.